ddtrace/contrib/flask/middleware.py

```diff
@@ -101,6 +101,14 @@ def _start_span(self):
                 service=self._service,
                 span_type=http.TYPE,
             )
+
+            # Distributed tracing
```

Comment:
I think this is a must have functionality, especially now that we're doing a lot of improvements for our distributed tracing compatibility. What I suggest here is having an approach similar to what we already have in our [aiohttp integration][1]. Mostly, it would be great if:
* we make distributed tracing configurable, so another kwarg [in the middleware][2] (let's say: `distributed_tracing=False`). In that way, distributed tracing can be enabled/disabled, and by default is disabled (most conservative approach)
* make it possible to set independently the `trace_id` and the `parent_id`. This is a wanted functionality, because when you're moving a trace from service A to service B, you want both fields. But in cases where the `trace_id` is (for instance) the request ID chosen by HAProxy, you want to create the root span of your trace with such ID.

Makes sense to you these changes? I think they improve your current implementation, while providing the same functionality you need. Thanks a lot for the contribution!


[1]: https://github.com/BrendanBall/dd-trace-py/blob/69f5fed4ac1cdb56a75636d2ad3a626cdd187f70/ddtrace/contrib/aiohttp/middlewares.py#L39-L48
[2]: https://github.com/BrendanBall/dd-trace-py/blob/69f5fed4ac1cdb56a75636d2ad3a626cdd187f70/ddtrace/contrib/flask/middleware.py#L25